dot/state/service.go

```diff
@@ -175,9 +192,27 @@ func (s *Service) Start() error {
 
 	// Start background goroutine to GC pruned keys.
 	go s.Storage.pruneStorage(s.closeCh)
+
+	if s.Metrics.Publish {
+		go s.updateMetrics()
+	}
 	return nil
 }
 
+func (s *Service) updateMetrics() {
+	ticker := time.NewTicker(s.Metrics.Interval)
+	for {
+		select {
+		case <-s.closeCh:
```

Comment:
use a context instead. Ideally pass it as argument to the function

Reply:
this service uses a `closeCh` to end spawned goroutines.